src/compiler/parser.ts

```diff
@@ -5089,6 +5104,18 @@ namespace ts {
             const fullStart = getNodePos();
             const decorators = parseDecorators();
             const modifiers = parseModifiers();
+            if (modifiers && modifiers.some(m => m.kind === SyntaxKind.DeclareKeyword)) {

```

Comment:
Could just do `if (some(modifiers, m => m.kind === SyntaxKind.DeclareKeyword))` as this does the existential check for you.